src/libsync/capabilities.cpp

```diff
@@ -211,7 +211,7 @@ bool Capabilities::isClientStatusReportingEnabled() const
 QList<QByteArray> Capabilities::supportedChecksumTypes() const
 {
     QList<QByteArray> list;
-    foreach (const auto &t, _capabilities["checksums"].toMap()["supportedTypes"].toList()) {
+    for (const auto &t : _capabilities["checksums"].toMap()["supportedTypes"].toList()) {
```

Comment:
For these transformations taking place in capabilities I would do the following
```suggestion
    const auto supportedTypes = _capabilities["checksums"].toMap()["supportedTypes"].toList();
    QList<QByteArray> list(suppertedTypes.count());
    for (const auto &t : supportedTypes) {
```